flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSubpartitionConsumerMemoryDataManager.java

```diff
@@ -43,6 +44,9 @@ public class HsSubpartitionConsumerMemoryDataManager implements HsDataView {
     @GuardedBy("consumerLock")
     private final Deque<HsBufferContext> unConsumedBuffers = new LinkedList<>();
 
+    @GuardedBy("consumerLock")
```

Comment:
This is guarded by a lock, why it also required as atomic?

Reply:
The write or read is not always be locked by `consumerLock` in all methods, such as `getBacklog()`. The `consumerLock` is removed currently.